feat(channels-dvr) Add TV Everywhere image option

fenichelar opened the pull request:
**Description**
<!--
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
-->
⚒️ Fixes  https://github.com/truecharts/charts/pull/4527#issuecomment-1332360794

**⚙️ Type of change**

- [x] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
<!--
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-->

**📃 Notes:**
https://github.com/truecharts/containers/pull/12503

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [x] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [x] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

xstar97 commented:
@stavros-k we don't need two `channels-dvr`, right?
@fenichelar since the docker image doesn't get updated at all...probably a good idea to use a image selector instead on the original `channels-dvr` here's a  chart that does exactly that. If you need a reference > https://github.com/truecharts/charts/tree/master/charts/stable/minecraft-java

edit: `channels-dvr` now has a image selector instead

fenichelar commented:
@Xstar97 There are two docker images, one with TV Everywhere support and one without. I didn't know that an image selector could be used. I will update this PR.

fenichelar committed:
Add TV Everywhere image to Channels DVR

stavros-k commented on charts/incubator/channels-dvr/questions.yaml:
```diff
@@ -4,6 +4,17 @@ portals:
+      default: image
+      enum:
+        - value: image
+          description: Latest
```
```suggestion
          description: Channels DVR
```

stavros-k replied:
latest isn't very clear description I think

fenichelar replied:
Done

stavros-k commented on charts/incubator/channels-dvr/questions.yaml:
```diff
@@ -4,6 +4,17 @@ portals:
+        - value: image
+          description: Latest
+        - value: tveImage
+          description: TV Everywhere
```
```suggestion
          description: Channels DVR with TV Everywhere
```

fenichelar replied:
Done

stavros-k commented on charts/incubator/channels-dvr/values.yaml:
```diff
@@ -3,6 +3,11 @@ image:
+tveImage:
+  repository: fancybits/channels-dvr
+  pullPolicy: Always
+  tag: tve@sha256:0fb54da902ae9bb872634e38a653f66c2698373ab25c587d79d41f24271077cb
```
```suggestion
  repository: tccr.io/truecharts/channels-dvr-tve
  pullPolicy: Always
  tag: latest@sha256:6edc4e63493996747bf07f5f58d4ec0e8bfc2ed22fe2c5c27706a7c720e6a8a1
```

fenichelar replied:
Is the TrueCharts mirror converting the `tve` tag to `latest`?

stavros-k replied:
Yes because it's under another "repo" `/chanels-dvr-tve`

fenichelar replied:
Where does the new hash come from?

fenichelar committed:
Update Channels DVR image descriptions

fenichelar committed:
Update Channels DVR tveImage repository

fenichelar committed:
Add CI config to Channel DVR

stavros-k reviewed (changes requested):
Your force push removed `imageSelector` from `values.yaml`
and also wiped the ci values. Which you replaced with different values that I had added initially.

Is there a reason behind all this?

fenichelar committed:
Fix Channels DVR tveImage tag issue

fenichelar commented:
> Your force push removed `imageSelector` from `values.yaml` and also wiped the ci values. Which you replaced with different values that I had added initially.
> 
> Is there a reason behind all this?

No, that was an accident. I didn't realize you had made the change when I rebased. I am trying to get everything back.

fenichelar commented:
@stavros-k Sorry, what should the `imageSelector` be?

stavros-k committed:
apply fixes

fenichelar commented:
@stavros-k Thank you

fenichelar committed:
Fix newline issue

fenichelar committed:
Fix tveImage hash

stavros-k committed:
bump

truecharts-admin commented:
This PR is locked to prevent necro-posting on closed PRs. Please create a issue or contact staff on discord if you want to further discuss this
